rancher2/resource_rancher2_auth_config_githubapp.go

```diff
@@ -0,0 +1,120 @@
+package rancher2
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	managementClient "github.com/rancher/rancher/pkg/client/generated/management/v3"
+)
+
+func resourceRancher2AuthConfigGithubApp() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceRancher2AuthConfigGithubAppCreate,
+		Read:   resourceRancher2AuthConfigGithubAppRead,
+		Update: resourceRancher2AuthConfigGithubAppUpdate,
+		Delete: resourceRancher2AuthConfigGithubAppDelete,
+		Schema: authConfigGithubAppFields(),
+	}
+}
+
+func resourceRancher2AuthConfigGithubAppCreate(d *schema.ResourceData, meta interface{}) error {
+	client, err := meta.(*Config).ManagementClient()
+	if err != nil {
+		return err
+	}
+
+	auth, err := client.AuthConfig.ByID(AuthConfigGithubAppName)
+	if err != nil {
+		return fmt.Errorf("[ERROR] Failed to get Auth Config %s: %s", AuthConfigGithubAppName, err)
+	}
+
+	log.Printf("[INFO] Creating Auth Config %s %s", AuthConfigGithubAppName, auth.Name)
+
+	authGithubApp, err := expandAuthConfigGithubApp(d)
+	if err != nil {
+		return fmt.Errorf("[ERROR] Failed expanding Auth Config %s: %s", AuthConfigGithubAppName, err)
+	}
+
+	// Checking if other auth config is enabled
+	if authGithubApp.Enabled {
+		err = meta.(*Config).CheckAuthConfigEnabled(AuthConfigGithubAppName)
+		if err != nil {
+			return fmt.Errorf("[ERROR] Checking to enable Auth Config %s: %s", AuthConfigGithubAppName, err)
```

Comment:
I think this message could be slightly improved to indicate what happened. 
```suggestion
			return fmt.Errorf("[ERROR] Checking if an Auth Config other than %s is enabled: %s", AuthConfigGithubAppName, err)
```